### PR TITLE
Migrated Enable Apache Kafka Connect Auto Restart

### LINKED ISSUE #378: Migrating Kafka Connect Auto Restart
# What should we add to our content collections?

(more information is useful here! Were you trying to do something particular?)

Migrating Kafka Connect Auto Restart article https://github.com/aiven/devportal/issues/319

# Is this something you would work on yourself?

[X] I plan to open a pull request for this

### PR DESCRIPTION
**Pull request by ftisiot**
# What changed, and why it matters

Migrated Kafka Connect Automatic Restart on Failures -> https://help.aiven.io/en/articles/5088396-kafka-connect-auto-restart-on-failures

Fixes #378

### CODE REVIEW

File: docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+Enable Apache Kafka Connect connectors auto restart on failures
+===============================================================
+
+In case of an Apache Kafka Connect connector failure, automatic task restart is generally not recommended. Proper investigation on the problem's cause should be performed before attempting to restart the failed connector to avoid experiencing similar problems in the future. However, sometimes a task can fail due to a rare problem like the dedicated node going out of memory due to a huge surge of data.
```
Comment (anelook) [inline docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:4]: I wonder if active voice is better than to have passive in this paragraph. It is a bit heavy to read right now.

If you have a failure in a Kafka connector, instead of immediately restarting it we recommend a proper investigation of the cause so that similar problems are avoided in the future. 

However....

Comment (anelook) [inline docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:4]: what happens when "However, sometimes a task can fail due to a rare problem like the dedicated node going out of memory due to a huge surge of data. " Is it when we can restart automatically? It feels like a specific action is missing for this cause of failure.

Comment (anelook) [inline docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:4]: And one note on "Apache Kafka Connect connector" - can we use "Kafka connector", when the context is clear?

File: docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
+Enable Apache Kafka Connect connectors auto restart on failures
+===============================================================
+
+In case of an Apache Kafka Connect connector failure, automatic task restart is generally not recommended. Proper investigation on the problem's cause should be performed before attempting to restart the failed connector to avoid experiencing similar problems in the future. However, sometimes a task can fail due to a rare problem like the dedicated node going out of memory due to a huge surge of data. 
+
+.. Note::
+
+    We observed cases when, using the Debezium source connector for PostgreSQL, the Apache Kafka connector stopped working after the PostgreSQL maintenance update, due to the PostgreSQL inability to create replication slots before failover. In such cases the connector automatic restart could be a valid solution to avoid the problem.
```
Comment (anelook) [inline docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:8]: "Debezium source connector for PostgreSQL, the Apache Kafka connector ", here is this Debezium source connector same as the Apache Kafka connector? If yes, I'd rephrase to avoid confusion to "We observed cases when the Debezium source connector for PostgreSQL stopped working because PostgreSQL was unable to create replication slots after the maintenance update of PostgreSQL. " Does it still have the same meaning?

Comment (anelook) [inline docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:8]: Also, I wonder what reader can do with this note? what is the action point?

File: docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+Enable connector automatic restart with Aiven console
+-----------------------------------------------------
+
+Aiven provides an option, applicable to all connectors, to enable automatic connector restart in case of edge situations. The following takes the update of an existing Debezium-PostgreSQL connector as an example:
```
Comment (anelook) [inline docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:13]: I'd move ", applicable to all connectors," to the end of the sentence or as a separate sentence 

"Aiven provides an option to enable automatic connector restart in case of edge situations. All our connectors support this option."

Comment (anelook) [inline docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:13]: "The following takes the update of an existing Debezium-PostgreSQL connector as an example: " do we need to sound so formal?

what if we change this to "For example, follow these steps to update of an existing Debezium-PostgreSQL connector:"

File: docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+1. In the `Aiven console <https://console.aiven.io/>`_ navigate to the Aiven for Apache Kafka or Aiven for Apache Kafka Connect service pages where the connector is defined.
+
+2. Select to the **Connectors** tab and click on the connector to be updated
+
+3. Select the **Aiven** tab
+
+4. set the **Automatic restart** option to ``True``
+
+.. image:: /images/products/kafka/kafka-connect/set-automatic-restart-true.png
```
Comment (anelook) [inline docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:23]: I like images, but since we agreed to use as little of them as possible I don't feel that this one is necessary. 

In many other places we describe interface and not show pictures for similar cases.

File: docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+   :alt: Aiven console with Automatic restart option set to True
+
+After enabling the automatic restart, when failures have been detected by our management platform, the connector is restarted.
```
Comment (anelook) [inline docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:26]: I'd rephrase a bit - "Once you enabled the automatic restart,  the connector will be restarted every time a failure is detected by our management platform".

**Review by anelook — CHANGES_REQUESTED** [review]
Nice to learn something new :)

I left some comments, they are mainly on phrasing explanations. It might be Monday morning and lack of coffee, but many phrasings were difficult to read for me, they seem to use overly complex language which is not necessary for these cases.

 I left some alternative ideas. Let me know what you think :)

File: docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst (hunk 1 above)
Comment (ftisiot) [reply docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:4]: Agreed, thanks!

Comment (ftisiot) [reply docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:4]: Added sentence

File: docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst (hunk 2 above)
Comment (ftisiot) [reply docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:8]: Thanks, fixed!

File: docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst (hunk 3 above)
Comment (ftisiot) [reply docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:13]: Done, thanks!

Comment (ftisiot) [reply docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:13]: I'm starting thinking I sound toooooo formal when writing :D

File: docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst (hunk 4 above)
Comment (ftisiot) [reply docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:23]: Agreed, in the original docs there were 2 of them, I removed 1, but it's still 1 too much!

File: docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst (hunk 5 above)
Comment (ftisiot) [reply docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:26]: Agreed, fixed

File: docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst (hunk 2 above)
Comment (ftisiot) [reply docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:8]: I believe the action point is in the last sentence: "In such cases the connector automatic restart could be a valid solution to avoid the problem."

File: docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst (hunk 1 above)
Comment (ftisiot) [reply docs/products/kafka/kafka-connect/howto/enable-automatic-restart.rst:4]: I would leave this as it is for now... it's a touchy topic...better safe than sorry

**Review by anelook — APPROVED** [review]
